### PR TITLE
improved type-hinting and adding type-hinting to workflow

### PR DESCRIPTION
**Pull request by oliverangelil**

### CODE REVIEW

File: src/scraping/tools.py [hunk 1]
```diff
@@ -51,7 +51,10 @@ def login(debug: bool = False) -> requests.session:
         # Retrieve necessary token - regex explanation:
         # search for all occurrences that match pattern within re.search
         # (.*?) will match any content. It is still unclear how "?" helps
-        pageToken = re.search(r'window\.__pageToken = "(.*?)";', main_site).group(1)
+        search = re.search(r'window\.__pageToken = "(.*?)";', main_site)
+        assert search is not None
```
Comment (oliverangelil) [inline src/scraping/tools.py:52]: without this solution mentioned here (https://stackoverflow.com/a/68471435/5392289), mypy complains `error: Item "None" of "Optional[Match[str]]" has no attribute "group"`

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 pandas==1.4.3
 psycopg2-binary==2.9.3
 requests==2.28.1
+types-requests==2.28.0
```
Comment (RicSegundo) [inline requirements.txt]: Do you have to install it locally also?
Why is this preferable vs the old solution of importing typing from the standard library?

**Review by RicSegundo — APPROVED** [review]
Is the new requests necessary to use mypy module for type checking?
Or it just complains about the standard library typing module, and that's why you changed?

**Comment by RicSegundo** [comment]
Fix for the "Definition of "__hash__" in base class "list" is incompatible with definition in base class "Enum"  [misc]" and 
"Definition of "__hash__" in base class "dict" is incompatible with definition in base class "Enum"  [misc]":
Add # type: ignore" after the enum class definition that has a dict or a list (this is a sad fix as it just ignores the error)
https://cdreimer.medium.com/fixing-the-incompatible-dunder-method-error-in-mypy-4d1ed4f43e5e

I actually improved this: replaced the List type for a Tuple, as we only need to read from it and not change it, so it should be immutable. That solved problem 1
For the Dict, I changed it to a MutableMapping, as explained here:
https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html
(under the _Standard “duck types”_)

**Comment by RicSegundo** [comment]
It took me a few hours but I think I understood all the problems that mypy was raising - all tests are passing now
Let's talk tomorrow night, to see if you want to merge it?

Sorry about having added/committed on top of your PR, I just need to do it to understand if the code would pass the tests and work on the droplet

**Pull request merged**